src/main/java/com/purbon/kafka/topology/roles/CCloudAclsProvider.java

```diff
@@ -80,27 +79,25 @@ private TopologyAclBinding translateIfNecessary(TopologyAclBinding binding) thro
             + "Service Account names into ID(s). At some point in time this will not be required anymore, "
             + "so you can configure this out by using ccloud.service_account.translation.enabled=false (true by default)");
 
-    String principal = binding.getPrincipal();
-    Long translatedPrincipalId = lookupServiceAccountId.get(principal);
-    if (translatedPrincipalId == null) { // Translation failed, so we can't continue
+    Principal principal = Principal.fromString(binding.getPrincipal());
```

Comment:
I like the use of a POJO to model the principal 😄 , much mode clean

Reply:
Thank you